bpf/gotracer/go_nethttp.c

```diff
@@ -1166,6 +1219,72 @@ int beyla_uprobe_netFdRead(struct pt_regs *ctx) {
     return 0;
 }
 
+SEC("uprobe/bodyRead")
+int beyla_uprobe_bodyRead(struct pt_regs *ctx) {
+    void *goroutine_addr = GOROUTINE_PTR(ctx);
+    bpf_dbg_printk("=== uprobe/proc body read goroutine === ");
+    go_addr_key_t g_key = {};
+    go_addr_key_from_id(&g_key, goroutine_addr);
+
+    // Get the address of the slice struct (p)
+    u64 body_addr = (u64)GO_PARAM2(ctx);
+
+    server_http_func_invocation_t *invocation =
+        bpf_map_lookup_elem(&ongoing_http_server_requests, &g_key);
+    if (!invocation) {
+        bpf_dbg_printk("can't find invocation info for server call");
+        return 0;
+    }
+    invocation->body_addr = body_addr;
+
+    return 0;
+}
+
+SEC("uprobe/bodyReadRet")
+int beyla_uprobe_bodyReadReturn(struct pt_regs *ctx) {
+    void *goroutine_addr = GOROUTINE_PTR(ctx);
+    bpf_dbg_printk("=== uprobe/proc body read returns goroutine === ");
+    go_addr_key_t g_key = {};
+    go_addr_key_from_id(&g_key, goroutine_addr);
+
+    u64 n = (u64)GO_PARAM1(ctx);
+
+    server_http_func_invocation_t *invocation =
+        bpf_map_lookup_elem(&ongoing_http_server_requests, &g_key);
+    if (!invocation) {
+        bpf_dbg_printk("can't find invocation info for server call");
+        return 0;
+    }
+    // content-type is set in invocation in ServeHTTP
+    bpf_dbg_printk("n is %d", n);
+    bpf_dbg_printk("content type is %s", invocation->content_type);
+
+    char body_buf[HTTP_BODY_MAX_LEN] = {};
+    if (n > 0 && invocation->body_addr) {
```

Comment:
```suggestion
    if (n <= 0 || !invocation->body_addr) {
      return 0;
    }
```

rationale: 
 - [F.56: Avoid unnecessary condition nesting](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f56-avoid-unnecessary-condition-nesting)
 - https://www.kernel.org/doc/html/next/process/coding-style.html